### DISCUSSION TITLE
Edge functions not recognizing import map

### DISCUSSION THREAD

**Discussion by eRuaro** [supabase/supabase#14887] [Questions]
I have two edge functions namely `create-payment-link` and `retrieve-payment-link` but the import map they're using is `/home/deno/flag_import_map.json`, and not the `import_map.json` file in the `functions` folder. Thus they're having a `x was not mapped in import map.` where x is a particular package they're using.

Both functions have these imports:
```
import { serve } from "server"
import * as cryptojs from "crypto-js";

import { config } from "dotenv";
```

And the contents of `import_map.json` is:
```
{
  "imports": {
    "server": "https://deno.land/std@0.168.0/http/server.ts",
    "crypto-js": "https://cdn.skypack.dev/crypto-js",
    "dotenv": "https://deno.land/x/dotenv/mod.ts"
  }
}
```

This happens when trying to serve them both locally with: `npx supabase functions serve --debug --env-file ./supabase/.env --import-map ./supabase/functions/import_map.json` 

Serving them one at a time with `npx supabase functions serve function-name --debug --env-file ./supabase/.env ` works though

**Comment by Pictor13** [comment]
Also having the issue.  
Did you manage to suppress all those debug warnings?

The message is coming from: https://deno.land/x/deno@v1.9.1/cli/import_map.rs?source=#L409

I didn't debug further, but it seems that it is applied to all the non-explicit dependencies (soft dependencies), also for deps that should not be explicit inside the import-map.

It is anyway something coming after updating to a recent version with `npm update supabase`.  
I suppose the devs are experimenting with debug messaging. Hopefully they will read this disccussion and use this feedback.
